packages/web-components/fast-foundation/src/disclosure/disclosure.ts

```diff
@@ -0,0 +1,117 @@
+import { attr, FASTElement } from "@microsoft/fast-element";
+/**
+ * A Disclosure Custom HTML Element.
+ * Based largely on the {@link https://w3c.github.io/aria-practices/#disclosure | disclosure element }.
+ *
+ * @public
+ */
+export class Disclosure extends FASTElement {
+    /**
+     * See {@link https://www.w3.org/WAI/PF/aria/roles#button} for more information
+     * @public
+     * @remarks
+     * HTML Attribute: aria-expanded
+     */
+    @attr({ attribute: "aria-expanded" })
+    public ariaExpanded: true | false;
+
+    /**
+     * Identifies the element (or elements) whose contents or presence are controlled by the current element.
+     *
+     * {@link https://www.w3.org/TR/wai-aria-1.1/#aria-controls}
+     * @public
+     * @remarks
+     * HTML Attribute: aria-controls
+     */
+    @attr({ attribute: "aria-controls" })
+    public ariaControls: string;
+
+    /**
+     * Determines if the element should show the extra content or not.
+     *
+     * @public
+     */
+    @attr({ mode: "boolean" })
+    public expanded: boolean;
+
+    /**
+     * Invoker title
+     *
+     * @public
+     */
+    @attr
+    public title: string;
+
+    /**
+     * @internal
+     */
+    public details: HTMLDetailsElement;
+
+    /**
+     * @internal
+     */
+    public connectedCallback(): void {
+        super.connectedCallback();
+        this.setup();
+    }
+
+    /**
+     * @internal
+     */
+    public disconnectedCallback(): void {
+        super.disconnectedCallback();
+        this.details.removeEventListener("toggle", this.onToggle);
+    }
+
+    constructor() {
+        super();
+        this.ariaExpanded = false;
+        this.ariaControls = `disclosure-content-${this.uuid()}`;
```

Comment:
I'd prefer to either tie this to an `id` passed by the user or a fixed id value. If this is in the shadow, this won't conflict and the ID should be unique to the Shadow DOM. No need to generate a UUID here IMO.

Reply:
It's in shadow DOM so will replace it with a fixed id.